pkg/meta/utils.go

```diff
@@ -476,3 +477,239 @@ func (m *baseMeta) FastGetSummary(ctx Context, inode Ino, summary *Summary, recu
 	}
 	return 0
 }
+
+func (t *TreeSummary) visitRoot(visit func(*TreeSummary)) {
+	for tree := t; tree != nil; tree = tree.parent {
+		visit(tree)
+	}
+}
+
+func (m *baseMeta) GetTreeSummary(ctx Context, root *TreeSummary, depth, topN uint8, strict bool) syscall.Errno {
+	var st syscall.Errno
+	if strict {
+		st = m.getTreeSummary(ctx, root)
```

Comment:
We don't need to details of tree beyond `depth`, use GetSummary for them.

it's better to merge this two together

Reply:
Done